src/subprocess-win32.cc

```diff
@@ -60,7 +60,7 @@ HANDLE Subprocess::SetupPipe(HANDLE ioport) {
 
   // Get the write end of the pipe as a handle inheritable across processes.
   HANDLE output_write_handle =
-      CreateFileA(pipe_name, GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
+      CreateFileW(Utf8ToWide(pipe_name).c_str(), GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
```

Comment:
You are converting pipe_name from Utf8, to Wide, twice. The conversion is expensive, as seen in your benchmarks.

Keep the first conversion around, and then re-use it.